pkg/dockerregistry/server/repositoryconfig.go

```diff
@@ -74,6 +78,11 @@ func newRepositoryConfig(ctx context.Context, options map[string]interface{}) (r
 	} else {
 		context.GetLogger(ctx).Infof("DEPRECATED: %q is deprecated, use the %q instead", DockerRegistryURLEnvVar, OpenShiftDefaultRegistryEnvVar)
 	}
+	rc.registryAddr, err = getStringOption(DockerRegistryURLEnvVarOption, "dockerregistryurl", rc.registryAddr, options)
```

Comment:
Only if rc.registryAddr has not filled yet.

Reply:
i would say that having this in configuration file overrides all env vars?

Reply:
Your code erases previous value no matter what. And environment variables usually overrides values in a config file.